### PR TITLE
Handle BG/FG edge cases

### PR DESCRIPTION
**Pull request by big-nacho**
This should improve edge cases and solve the failing requests issue

### CODE REVIEW

File: src/process_binary.py [hunk 1]
```diff
@@ -44,43 +45,97 @@ def threshold_simple (img_grayscale):
...
+    bitmap, inverted_bitmap = [
+        np.where(candidate == 255, 0, 1).astype(np.uint32) 
+        for candidate in candidates
+    ]
+
+    pixel_count = get_pixel_count(img)
+    foreground_area = np.sum(bitmap)
+    inverted_foreground_area = np.sum(inverted_bitmap)
+
+    if (foreground_area / pixel_count) < 0.001:
```
Comment (phil-r) [inline src/process_binary.py:113]: `0.001` is not one percent

File: src/process_binary.py [hunk 2]
```diff
@@ -44,43 +45,97 @@ def threshold_simple (img_grayscale):
...
+        # If total area of the foreground is less than 1 percent of the image's
+
+        # It's possible that the inverted image produced a better bitmap.
+        # The main relevant case is an all-or-mostly white foreground with a transparent background.
+
+        #NOTE: The inverted bitmap CAN be worse than the non-inverted one.
+        # e.g A very small amount of black pixels on a transparent background
+
+        if inverted_foreground_area > foreground_area:
+            # Then the inverted bitmap is better. And its area is at least 1.
```
Comment (phil-r) [inline src/process_binary.py:123]: 1 what? pixel?

**Review by phil-r — CHANGES_REQUESTED** [review]

File: src/process_binary.py (hunk 2 above)
Comment (big-nacho) [reply src/process_binary.py:123]: Just 1. No unit applies here. These are just 1/0 bitmaps

**Review by phil-r — COMMENTED** [review]
lgtm

**Review by phil-r — APPROVED** [review]

**Pull request merged**